File: src/main/java/net/nodus/database/account/UserAccountRepository.java
package net.nodus.database.account;

import java.util.Optional;
import java.util.UUID;
import org.springframework.data.jpa.repository.JpaRepository;

public interface UserAccountRepository extends JpaRepository<UserAccount, UUID> {

    Optional<UserAccount> findByProviderAndProviderId(OAuthProvider provider,
        String providerId);
}

File: src/main/kotlin/net/nodus/account/UserAccountRepository.kt
package net.nodus.account

import org.springframework.data.mongodb.repository.MongoRepository
import java.time.Instant

interface UserAccountRepository : MongoRepository<UserAccount, String> {
    fun findByProviderAndProviderIdAndDeletedAtIsNull(
        provider: OAuthProvider,
        providerId: String,
    ): UserAccount?

    fun findByDeletedAtBefore(deletedAt: Instant): List<UserAccount>
}
